src/post.js

```diff
@@ -10,12 +10,42 @@ module.exports = async config => {
   const name = `v${pkg.version}`;
   const tag = {owner, repo, ref: `refs/tags/${name}`, sha: await promisify(gitHead)()};
   const body = await promisify(plugins.generateNotes)(config);
-  const release = {owner, repo, tag_name: name, name, target_commitish: branch, draft: !!debug, body};
+  let githubPreRelease = false;
+  let release = {
+    owner,
+    repo,
+    tag_name: name,
+    name,
+    target_commitish: branch,
+    draft: !!debug,
+    body,
+    prerelease: githubPreRelease,
+  };
 
   if (debug && !githubToken) {
     return {published: false, release};
   }
 
+  // obtain npm dist tag if any
+  let npmDistTag = (pkg.publishConfig || {}).tag;
+
+  // if npmDistTag is set, we can consider github pre release
+  if (npmDistTag != null) {
+    // obtain release.no-pre-release config
+    const noPreReleaseTags = (pkg.release || {})['no-pre-release'] || ['latest'];
```

Comment:
You could use Object destructuring to get the value of `pkg.release` in [src/post.js#L8](https://github.com/semantic-release/semantic-release/blob/caribou/src/post.js#L8) like it's done for other input params.

Reply:
Not sure how to do it here. Can you help?

Reply:
I was thinking something like: `const {pkg: {release} = {}, options: {branch, debug, githubUrl, githubToken, githubApiPathPrefix}, plugins} = config;`
But actually, I just realized `pkg` is used somewhere else so nevermind.